### PR TITLE
feat: add authentication

### PR DESCRIPTION
**Pull request by anik31**
## Authentication
Added:
- sign in
- sign up
- logout
 
## Files to check
- src/components/RequireAuth.js
- src/components/navbar/Navbar.jsx
- src/context/auth-context.js
- src/pages/auth/Login.jsx
- src/pages/auth/Signup.jsx
 
 _PS - You can review the CSS files if you want to. Ignore other files._

https://user-images.githubusercontent.com/56336326/162265324-d1fbbc6c-e4bd-445d-a29b-4cdacc798fae.mp4

### CODE REVIEW

**Review by VLeads — APPROVED** [review]
Looking good. You can merge

File: src/pages/auth/Signup.jsx [hunk 1]
```diff
@@ -1,41 +1,56 @@
...
+    const passwordVisibilityHandler = (e) => setIsPasswordVisible(prev=> !prev);
 
-    const confirmPasswordVisibilityHandler = (e) => {
-        e.preventDefault();
-        setIsConfirmPasswordVisible(prev=> !prev);
-    }
+    const confirmPasswordVisibilityHandler = (e) => setIsConfirmPasswordVisible(prev=> !prev);
     
+    const signUpHandler = () => {
+        if(!credentials.email || !credentials.password || !credentials.firstName || !credentials.lastName || !confirmPass){
```
Comment (sainath7878) [inline src/pages/auth/Signup.jsx:24]: Appreciation: Good that you're validating fields!

**Review by sainath7878 — APPROVED** [review]
Code looks neat and clean. Have nothing to suggest. Great work!

File: src/pages/auth/Signup.jsx (hunk 1 above)
Comment (anik31) [reply src/pages/auth/Signup.jsx:24]: thanks

**Pull request merged**